schema_based_on_diagram.sql

```diff
@@ -0,0 +1,56 @@
+--Create Patients Table
+CREATE TABLE patients (
+    id INTEGER NOT NULL,
+    name VARCHAR(50) NOT NULL,
+    date_of_birth DATE NOT NULL
+);
+ALTER TABLE
+    patients ADD PRIMARY KEY(id);
+-- Create medical history table
+CREATE TABLE medical_histories (
+    id INTEGER NOT NULL,
+    admitted_at TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    patient_id INTEGER NOT NULL,
+    status VARCHAR(255) NOT NULL
+);
+ALTER TABLE
+    medical_histories ADD PRIMARY KEY(id);
+--Create treatment table
+CREATE TABLE treatments (
+    id INTEGER NOT NULL,
+    type VARCHAR(255) NOT NULL,
+    name VARCHAR(255) NOT NULL
+);
+ALTER TABLE
+    treatments ADD PRIMARY KEY(id);
+-- Create invoice table
+CREATE TABLE invoices (
+    id INTEGER NOT NULL,
+    total_amount DECIMAL(8, 2) NOT NULL,
+    generated_at TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    payed_at TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    medical_history__id INTEGER NOT NULL
+);
+ALTER TABLE
+    invoices ADD PRIMARY KEY(id);
+-- Create invoice_items Table
+CREATE TABLE invoice_items (
+    id INTEGER NOT NULL,
+    unit_price DECIMAL(8, 2) NOT NULL,
+    quantity INTEGER NOT NULL,
+    total_price DECIMAL(8, 2) NOT NULL,
+    invoice_id INTEGER NOT NULL,
+    treatment_id INTEGER NOT NULL
+);
+
+
+ALTER TABLE
+    invoice_items ADD PRIMARY KEY(id);
+ALTER TABLE
+    medical_histories ADD CONSTRAINT medical_histories_patient_id_foreign FOREIGN KEY(patient_id) REFERENCES patients(id);
+ALTER TABLE
+    invoice_items ADD CONSTRAINT invoice_items_invoice_id_foreign FOREIGN KEY(invoice_id) REFERENCES invoices(id);
+ALTER TABLE
+    invoices ADD CONSTRAINT invoices_medical_history__id_foreign FOREIGN KEY(medical_history__id) REFERENCES medical_histories(id);
+ALTER TABLE
```

Comment:
- [ ] Kindly create indexes for all foreign keys

Reply:
Done. Thanks for the review @AshakaE